src/ninja.cc

```diff
@@ -1118,15 +1118,79 @@ int ReadFlags(int* argc, char*** argv,
   return -1;
 }
 
+int ReadFlagsFromCommandLine(int* argc, char*** argv,
+              Options* options, BuildConfig* config) {
+  int exit_code = ReadFlagsFromArray(argc, argv, options, config);
+  if ( exit_code == -1 ) {
+    *argv += optind;
+    *argc -= optind;
+  }
+  return exit_code;
+}
+
+int ReadFlagsFromEnv(Options* options, BuildConfig* config) {
+  char *str = getenv("NINJA_FLAGS");
+  if (str == NULL || strlen(str) == 0 )
+    return -1;
+
+  char *str_copy = (char *)malloc(strlen(str)+1);
+  char **argv = (char **)calloc(strlen(str), sizeof(char *));
+  int argc = 0, i = 0;
+  bool in_arg = false;
+
+  argv[argc++] = (char *)""; // fake argv[0] (command) for getopt
```

Comment:
travis is failing because

```
src/ninja.cc:1141:16: warning: deprecated conversion from string constant to ‘char*’ [-Wwrite-strings]
   argv[argc++] = ""; // fake argv[0] (command) for getopt
                ^
```

(click the red x to get to that: https://travis-ci.org/ninja-build/ninja/builds/354768286?utm_source=github_status&utm_medium=notification)